assets/js/components/legacy-notifications/util.js

```diff
@@ -87,10 +88,15 @@ const removeDismissed = ( notifications ) => {
 		return notifications;
 	}
 
-	return notifications.filter( ( notification ) => {
-		const dismissed = getCache( `notification::dismissed::${ notification.id }` );
+	const promises = await notifications.map( ( notification ) => getItem( `notification::dismissed::${ notification.id }` ) );
```

Comment:
Probably an oversight but these only need to be awaited below by `Promise.all`.
```suggestion
	const promises = notifications.map( ( notification ) => getItem( `notification::dismissed::${ notification.id }` ) );
```